client/lclusterc.go

```diff
@@ -0,0 +1,324 @@
+/*
+ * File: client/main.go
+ *
+ * Description: Contains the lcluster client code.
+ *
+ * TODO: removejob is broken, others funcs are unimplemented
+ */
+
+package main
+
+import (
+	pb "../lclusterpb"
+	"flag"
+	"fmt"
+	"golang.org/x/net/context"
+	grpc "google.golang.org/grpc"
+	"strconv"
+	"time"
+)
+
+// Variables to hold the result of argument flags.
+var addjob string
+var checkjob string
+var removejob string
+
+// Initialize the flags beforehand.
+func init() {
+
+	// Argument flag for when end-user requests to add a job
+	flag.StringVar(&addjob, "addjob", "",
+		"Commandline program to execute; e.g. 'grep abc /path/to/file' ")
+
+	// Argument flag for when the end-user wants to check on a job
+	flag.StringVar(&checkjob, "checkjob", "",
+		"Uuid of the job to query status.")
+
+	// Argument flag for when the end-user wants to remove a job
+	flag.StringVar(&removejob, "removejob", "",
+		"Pid of the job to be removed.")
+}
+
+//! Add a job to the server.
+/*
+ * @param    string    commandline to execute
+ *
+ * @return   none
+ */
+func addJobToServer(cmd string) {
```

Comment:
Done, moved the client logic out to its own separate library.

https://github.com/rbisewski/lclusterd/commit/e360704cb5bb246cde6d1c029996860c2a0c642c